ntbs-service/Pages/Notifications/LegacyView.cshtml.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using ntbs_service.DataMigration;
+using ntbs_service.Models;
+using ntbs_service.Services;
+
+namespace ntbs_service.Pages.Notifications
+{
+    public class LegacyView : PageModel
+    {
+        private readonly ILegacySearchService _legacySearchService;
+        private readonly INotificationImportService _notificationImportService;
+
+        [BindProperty] 
+        public NotificationBannerModel NotificationBanner { get; set; }
+        
+        [BindProperty(SupportsGet = true)]
+        public string LegacyNotificationId { get; set; }
+        
+        public string RequestId { get; set; }
+        public ImportResult LegacyImportResult { get; set; }
+
+        public LegacyView(ILegacySearchService legacySearchService, INotificationImportService notificationImportService)
+        {
+            _legacySearchService = legacySearchService;
+            _notificationImportService = notificationImportService;
+        }
+        
+        public async Task<IActionResult> OnGetAsync()
+        {
+            await GetLegacyNotificationDetailsForBanner();
+            return Page();
+        }
+
+        private async Task GetLegacyNotificationDetailsForBanner()
+        {
+            NotificationBanner = await _legacySearchService.SearchByIdAsync(LegacyNotificationId);
+            NotificationBanner.ShowLink = false;
```

Comment:
Happy to be ignored on this one but I think I'd prefer the constructor of the notification banner model in legacySearchService to have a flag for ShowLink instead of manually setting it here, not sure what is cleaner

Reply:
Do you having a method that sets showLink to false?
Otherwise I will need to pass parameter into SearchByIdAsync method which sets it in constructor.

Reply:
SearchByIdAsync just wants it to be false always right? I was thinking more along the lines of AsNotificationBannerAsync(dynamic result, bool showLink = true) and then SearchByIdAsync would cool AsNotificationBannerAsync(result, false). Again happy to be ignored just what I thought might be neater

Reply:
This seems like we will tie showLink = false to SearchByIdAsync which should really care about whether flag is true or false

Reply:
Sure, probably not worth it then :)